### ISSUE TITLE
Unable to parse sqlite `on conflict` clause

### ISSUE THREAD

**Issue by hallettj** [sqlfluff/sqlfluff#3872]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff is a very helpful linter! But unless I'm missing something it seems that it does not support `on conflict` clauses in SQLite queries. When reading SQLite queries that contain an `on conflict` clause SQLFluff reports errors of the form,

> Found unparsable section: 'ON CONFLICT(word) DO UPDATE SET count=co...

I have tested with an upsert, and with a unique index. This example copied from [SQLite's upsert documentation](https://www.sqlite.org/lang_UPSERT.html) does not parse:

```sql
INSERT INTO vocabulary(word) VALUES('jovial')
ON CONFLICT(word) DO UPDATE SET count=count+1;
```

This minimal create table example similarly does not parse:

```sql
create table imap_boxes (
    account_id integer not null,
    box_name text not null,
    unique (account_id, box_name) on conflict replace
);
```

There are details on where `on conflict` clauses may appear in [SQLite's ON CONFLICT documentation](https://www.sqlite.org/lang_conflict.html).

### Expected Behaviour

I expected to SQLFluff to not report any "unparsable section" errors.

### Observed Behaviour

The upsert example results in the error:

> PRS: Line 2, Position 1: Found unparsable section: 'ON CONFLICT(word) DO UPDATE SET count=co...

The create table example results in the error:

> PRS: Line 1, Position 1: Found unparsable section: 'create table imap_boxes (\n account_id...

These "unparsable section" errors go away if I comment out the `on conflict` clause.

### How to reproduce

Check this sql:

```sql
create table imap_boxes (
    account_id integer not null,
    box_name text not null,
    unique (account_id, box_name) on conflict replace
);
```

with this `.sqlfluff` config:

```cfg
[sqlfluff]
dialect = sqlite
```

### Dialect

sqlite

### Version

sqlfluff, version 1.3.1

Python 3.10.6

### Configuration

```cfg
[sqlfluff]
dialect = sqlite
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by dcecile** [comment]
Same issue with another upsert form:

```sh
echo "insert into version (id) values (cast(:id as integer)) on conflict do nothing;" >script.sql        
sqlfluff lint script.sql --dialect=sqlite
```

Is there any workaround?

**Comment by Enduriel** [comment]
This was added to the postgres dialect with #3027 but indeed still doesn't work for sqlite.
